packages/remix-dev/config/defaults/entry.server.node.tsx

```diff
@@ -15,51 +15,13 @@ export default function handleRequest(
   remixContext: EntryContext,
   loadContext: AppLoadContext
 ) {
-  let prohibitOutOfOrderStreaming =
-    isBotRequest(request.headers.get("user-agent")) || remixContext.isSpaMode;
-
-  return prohibitOutOfOrderStreaming
-    ? handleBotRequest(
-        request,
-        responseStatusCode,
-        responseHeaders,
-        remixContext
-      )
-    : handleBrowserRequest(
-        request,
-        responseStatusCode,
-        responseHeaders,
-        remixContext
-      );
-}
-
-// We have some Remix apps in the wild already running with isbot@3 so we need
-// to maintain backwards compatibility even though we want new apps to use
-// isbot@4.  That way, we can ship this as a minor Semver update to @react-router/dev.
-function isBotRequest(userAgent: string | null) {
-  if (!userAgent) {
-    return false;
-  }
-
-  // isbot >= 3.8.0, >4
-  if ("isbot" in isbotModule && typeof isbotModule.isbot === "function") {
-    return isbotModule.isbot(userAgent);
-  }
-
-  // isbot < 3.8.0
-  if ("default" in isbotModule && typeof isbotModule.default === "function") {
-    return isbotModule.default(userAgent);
-  }
+  let userAgent = request.headers.get("user-agent");
 
-  return false;
-}
+  // Ensure requests from bots and SPA Mode renders wait for all content to load before responding
+  // https://react.dev/reference/react-dom/server/renderToPipeableStream#waiting-for-all-content-to-load-for-crawlers-and-static-generation
+  let prohibitOutOfOrderStreaming =
+    (userAgent && isbot(userAgent)) || remixContext.isSpaMode;
```

Comment:
Just for my learning, does the checking for `isSpaMode` here negate the need for a separate [entry.server.spa.tsx](https://github.com/remix-run/react-router/blob/9dcd08ba625656d3f57efbf88b8729b22d64422a/packages/remix-dev/config/defaults/entry.server.spa.tsx) file?

Reply:
With turbo-stream we have to do away with any "renderToString" and use the async version with buffering going forward.

Reply:
yeah we have a roadmap item to remove that for v7 https://github.com/remix-run/react-router/issues/11642